Source/Protocols/Request.swift

```diff
@@ -6,24 +6,32 @@ public typealias EmptyResponse = Void
 
 public protocol Request {
   typealias ResponseObject
+  typealias ResponseError = NSError
   func build() -> NSURLRequest
-  func parse(j: JSON) -> Result<ResponseObject, NSError>
+  func parse(j: JSON) -> Result<ResponseObject, SwishError>
+  func transformError(error: SwishError) -> ResponseError
```

Comment:
I'm unclear on where users would be able to actually _use_ that method. It'd move the logic outside the request and into some object up above the `APIClient`.